Source/ACE.Server/WorldObjects/Player_Inventory.cs

```diff
@@ -291,13 +291,26 @@ public bool TryEquipObjectWithNetworking(WorldObject item, EquipMask wieldedLoca
                     if (item.HasProcSpell((uint)spell.Spell))
                         continue;
 
-                    if (spell.Probability == 0.0f)
+                    if (ProcSpell.HasValue && spell.Spell == ProcSpell.Value)
```

Comment:
this is checking ProcSpell / SpellDID on the player, instead of the item

should just be

```
if (spell.Spell == item.SpellDID || spell.Spell == item.ProcSpell)
    continue;
```

the logging is unneeded, as we now have a query that finds all of these cases across the entire db